### ISSUE TITLE
Flight client doesn't publish Boolean column

### ISSUE THREAD

**Issue by ilyaberdnikov** [deephaven/deephaven-core#4005]
**Description**

A small modification of `AggInputBase` fails with an error.

**Steps to reproduce**
1. Apply patch
```
diff --git a/java-client/flight-examples/src/main/java/io/deephaven/client/examples/AggInputBase.java b/java-client/flight-examples/src/main/java/io/deephaven/client/examples/AggInputBase.java
index 6c882ef15..0684aa668 100644
--- a/java-client/flight-examples/src/main/java/io/deephaven/client/examples/AggInputBase.java
+++ b/java-client/flight-examples/src/main/java/io/deephaven/client/examples/AggInputBase.java
@@ -8,7 +8,7 @@ import io.deephaven.client.impl.TableHandle;
 import io.deephaven.qst.LabeledValue;
 import io.deephaven.qst.LabeledValues;
 import io.deephaven.qst.column.header.ColumnHeader;
-import io.deephaven.qst.column.header.ColumnHeaders8;
+import io.deephaven.qst.column.header.ColumnHeaders9;
 import io.deephaven.qst.table.InMemoryKeyBackedInputTable;
 import io.deephaven.qst.table.LabeledTable;
 import io.deephaven.qst.table.LabeledTables;
@@ -50,6 +50,7 @@ abstract class AggInputBase extends FlightExampleBase {
     protected void execute(FlightSession flight) throws Exception {
         final ColumnHeader<Integer> inputKey = INPUT_KEY;
         final ColumnHeader<Integer> groupKey = GROUP_KEY;
+        final ColumnHeader<Boolean> b = ColumnHeader.ofBoolean("B");
         final ColumnHeader<Byte> u = ColumnHeader.ofByte("U");
         final ColumnHeader<Short> v = ColumnHeader.ofShort("V");
         final ColumnHeader<Integer> w = ColumnHeader.ofInt("W");
@@ -59,7 +60,7 @@ abstract class AggInputBase extends FlightExampleBase {
 
         final Builder builder = LabeledTables.builder();
 
-        final TableSpec spec = InMemoryKeyBackedInputTable.of(TableHeader.of(inputKey, groupKey, u, v, w, x, y, z),
+        final TableSpec spec = InMemoryKeyBackedInputTable.of(TableHeader.of(inputKey, groupKey, b, u, v, w, x, y, z),
                 Collections.singletonList(inputKey.name()));
         builder.putMap("base", spec);
         for (LabeledTable labeledTable : labeledTables(spec)) {
@@ -72,8 +73,8 @@ abstract class AggInputBase extends FlightExampleBase {
                 flight.session().publish(result.name(), result.value()).get(5, TimeUnit.SECONDS);
             }
 
-            final ColumnHeaders8<Integer, Integer, Byte, Short, Integer, Long, Float, Double> headers =
-                    inputKey.header(groupKey).header(u).header(v).header(w).header(x).header(y).header(z);
+            final ColumnHeaders9<Integer, Integer, Boolean, Byte, Short, Integer, Long, Float, Double> headers =
+                    inputKey.header(groupKey).header(b).header(u).header(v).header(w).header(x).header(y).header(z);
 
             final TableHandle base = results.get("base");
 
@@ -82,20 +83,21 @@ abstract class AggInputBase extends FlightExampleBase {
             final int sizeGuess = (int) Math.round(inputTableSize * updatePercentage);
 
             while (true) {
-                final ColumnHeaders8<Integer, Integer, Byte, Short, Integer, Long, Float, Double>.Rows rows =
+                final ColumnHeaders9<Integer, Integer, Boolean, Byte, Short, Integer, Long, Float, Double>.Rows rows =
                         headers.start(sizeGuess);
                 for (int i = 0; i < inputTableSize; ++i) {
                     if (random.nextDouble() > updatePercentage) {
                         continue;
                     }
                     final int group = random.nextInt(numGroups);
+                    final boolean b_ = random.nextBoolean();
                     final byte u_ = (byte) random.nextInt();
                     final short v_ = (short) random.nextInt();
                     final int w_ = random.nextInt();
                     final long x_ = random.nextLong();
                     final float y_ = random.nextFloat();
                     final double z_ = random.nextDouble();
-                    rows.row(i, group, u_, v_, w_, x_, y_, z_);
+                    rows.row(i, group, b_, u_, v_, w_, x_, y_, z_);
                 }
                 flight.addToInputTable(base, rows.newTable(), bufferAllocator);
                 Thread.sleep(sleepMillis);

```
2. Run [io.deephaven.client.examples.AggByExample](https://github.com/deephaven/deephaven-core/blob/main/java-client/flight-examples/src/main/java/io/deephaven/client/examples/AggByExample.java)

**Expected results**

A table with column `B` full of random Booleans would appear.

**Actual results**

```
heduler-Concurrent-1 | i.d.s.s.SessionState      | Internal Error '791d3b18-1c51-476e-8491-34add9ee78fa' java.lang.RuntimeException: NewValue key and key don't match. key def=io.deephaven.base.Pair@7bcb0692, key=io.deephaven.base.Pair@e5badb20
        at io.deephaven.hash.KeyedObjectHash.checkSameKey(KeyedObjectHash.java:1144)
        at io.deephaven.hash.KeyedObjectHash.internalCreateIfAbsent(KeyedObjectHash.java:1068)
        at io.deephaven.hash.KeyedObjectHash.putIfAbsent(KeyedObjectHash.java:923)
        at io.deephaven.engine.table.impl.sources.NullValueColumnSource.getInstance(NullValueColumnSource.java:48)
        at io.deephaven.engine.table.impl.sources.NullValueColumnSource.lambda$createColumnSourceMap$1(NullValueColumnSource.java:55)
        at java.util.stream.Collectors.lambda$toMap$68(Collectors.java:1674)
        at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
        at io.deephaven.engine.table.impl.sources.NullValueColumnSource.createColumnSourceMap(NullValueColumnSource.java:53)
        at io.deephaven.engine.table.impl.util.KeyedArrayBackedMutableTable.make(KeyedArrayBackedMutableTable.java:54)
        at io.deephaven.server.table.ops.CreateInputTableGrpcImpl.create(CreateInputTableGrpcImpl.java:76)
        at io.deephaven.server.table.ops.CreateInputTableGrpcImpl.create(CreateInputTableGrpcImpl.java:29)
        at io.deephaven.server.table.ops.TableServiceGrpcImpl$BatchExportBuilder.doExport(TableServiceGrpcImpl.java:689)
        at io.deephaven.server.table.ops.TableServiceGrpcImpl.lambda$batch$5(TableServiceGrpcImpl.java:541)
        at io.deephaven.server.session.SessionState$ExportObject.doExport(SessionState.java:897)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
        at java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at io.deephaven.server.runner.DeephavenApiServerModule$ThreadFactory.lambda$newThread$0(DeephavenApiServerModule.java:174)
        at org.jpy.PyLib.callAndReturnObject(PyLib.java:-2)
        at org.jpy.PyObject.call(PyObject.java:449)
        at io.deephaven.server.console.python.DebuggingInitializer.lambda$createInitializer$0(DebuggingInitializer.java:30)
        at java.lang.Thread.run(Thread.java:833)
```
**Additional details and attachments**

Similarly, another minimal example:

```
         TableHeader header = TableHeader.of(
                 ColumnHeader.ofBoolean("Bool"),
                 ColumnHeader.ofInt("Int"),
                 ColumnHeader.ofString("String")
         );

         TableHandle handle = flightSession.session().execute(InMemoryAppendOnlyInputTable.of(header));
```
this call to `execute` throws `TableHandle.TableHandleException` and the IDE shows pretty much the same

```
heduler-Concurrent-4 | i.d.s.s.SessionState      | Internal Error '1e55479f-951e-45cf-9b86-47d7ebaf8854' java.lang.RuntimeException: NewValue key and key don't match. key def=io.deephaven.base.Pair@7bcb0692, key=io.deephaven.base.Pair@e5badb20
        at io.deephaven.hash.KeyedObjectHash.checkSameKey(KeyedObjectHash.java:1144)
        at io.deephaven.hash.KeyedObjectHash.internalCreateIfAbsent(KeyedObjectHash.java:1068)
        at io.deephaven.hash.KeyedObjectHash.putIfAbsent(KeyedObjectHash.java:923)
        at io.deephaven.engine.table.impl.sources.NullValueColumnSource.getInstance(NullValueColumnSource.java:48)
        at io.deephaven.engine.table.impl.sources.NullValueColumnSource.lambda$createColumnSourceMap$1(NullValueColumnSource.java:55)
        at java.util.stream.Collectors.lambda$toMap$68(Collectors.java:1674)
        at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
        at io.deephaven.engine.table.impl.sources.NullValueColumnSource.createColumnSourceMap(NullValueColumnSource.java:53)
        at io.deephaven.engine.table.impl.util.AppendOnlyArrayBackedMutableTable.make(AppendOnlyArrayBackedMutableTable.java:55)
        at io.deephaven.engine.table.impl.util.AppendOnlyArrayBackedMutableTable.make(AppendOnlyArrayBackedMutableTable.java:40)
        at io.deephaven.server.table.ops.CreateInputTableGrpcImpl.create(CreateInputTableGrpcImpl.java:74)
        at io.deephaven.server.table.ops.CreateInputTableGrpcImpl.create(CreateInputTableGrpcImpl.java:29)
        at io.deephaven.server.table.ops.TableServiceGrpcImpl$BatchExportBuilder.doExport(TableServiceGrpcImpl.java:689)
        at io.deephaven.server.table.ops.TableServiceGrpcImpl.lambda$batch$5(TableServiceGrpcImpl.java:541)
        at io.deephaven.server.session.SessionState$ExportObject.doExport(SessionState.java:897)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
        at java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at io.deephaven.server.runner.DeephavenApiServerModule$ThreadFactory.lambda$newThread$0(DeephavenApiServerModule.java:174)
        at org.jpy.PyLib.callAndReturnObject(PyLib.java:-2)
        at org.jpy.PyObject.call(PyObject.java:449)
        at io.deephaven.server.console.python.DebuggingInitializer.lambda$createInitializer$0(DebuggingInitializer.java:30)
        at java.lang.Thread.run(Thread.java:833)
```

**Versions**
 - Deephaven: 0.25.2
 - OS: MacOs

**Comment by lbooker42** [comment]
I've investigated this and there is a mismatch between the table definition created by the GRPC when a boolean column is created and the underlying column sources created on the server.

Briefly, the ColumnDefinition parsed by `io.deephaven.server.table.ops.CreateInputTableGrpcImpl#create` is `boolean` type but we effectively don't allow primitive `boolean` sources to be created on the server.  The `AbstractColumnSource` constructor automatically upcasts to `Boolean` (and `NullColumnSource` subclasses) so multiple comparisons between the definition and the created column fail.

**Comment by lbooker42** [comment]
This is fixed by #4047. Please note that some aggregations do not accept a `Boolean` input column.  In the list of aggregations in `AggByExample.java`, attempting to use `avgBy`, `stdBy`, `varBy` will result in an `UnsupportedOperationException` if applied to a boolean column source as in the reproducing code.
